edb/server/pgcluster.py

```diff
@@ -1104,3 +985,220 @@ async def _get_reserved_connections(
         pg_config_path=str(pg_config),
         instance_params=instance_params,
     )
+
+
+async def _run_logged_text_subprocess(
+    args: Sequence[str],
+    logger: logging.Logger,
+    level: int = logging.DEBUG,
+    check: bool = True,
+    log_stdout: bool = True,
+    timeout: Optional[float] = None,
+    **kwargs: Any,
+) -> Tuple[List[str], List[str], int]:
+    stdout_lines, stderr_lines, exit_code = await _run_logged_subprocess(
+        args,
+        logger=logger,
+        level=level,
+        check=check,
+        log_stdout=log_stdout,
+        timeout=timeout,
+        **kwargs,
+    )
+
+    return (
+        [line.decode() for line in stdout_lines],
+        [line.decode() for line in stderr_lines],
+        exit_code,
+    )
+
+
+async def _run_logged_subprocess(
+    args: Sequence[str],
+    logger: logging.Logger,
+    level: int = logging.DEBUG,
+    check: bool = True,
+    log_stdout: bool = True,
+    log_stderr: bool = True,
+    capture_stdout: bool = True,
+    capture_stderr: bool = True,
+    timeout: Optional[float] = None,
+    **kwargs: Any,
+) -> Tuple[List[bytes], List[bytes], int]:
+    process, stdout_reader, stderr_reader = await _start_logged_subprocess(
+        args,
+        logger=logger,
+        level=level,
+        log_stdout=log_stdout,
+        log_stderr=log_stderr,
+        capture_stdout=capture_stdout,
+        capture_stderr=capture_stderr,
+        **kwargs,
+    )
+
+    exit_code, stdout_lines, stderr_lines = await asyncio.wait_for(
```

Comment:
Maybe use taskgroup?

Reply:
Why is a taskgroup better here?  `wait_for` shouldn't leak tasks, no?

Reply:
> Why is a taskgroup better here?

More predictable failing than gather if something goes wrong?

Reply:
Up to you, the current code is ok too